File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {

    let map = {};
    let maxV = maxC = 0;
    const vowels = ['a','e','i','o','u'];

    for(let i=0;i<s.length;i++) {

        if(!map[s[i]]) map[s[i]]=1;
        else map[s[i]]++;
    }

    const mapKeys = Object.keys(map);
    
    for(let i=0;i<mapKeys.length;i++) {

        if(vowels.includes(mapKeys[i])) maxV = Math.max(maxV,map[mapKeys[i]]);
        else maxC = Math.max(maxC,map[mapKeys[i]]);
    }

    return maxV+maxC;  
};

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {

        int maxV = 0,maxC = 0;
        Map<Character,Integer> map = new HashMap<>();

        for(int i=0;i<s.length();i++) {
            char x = s.charAt(i);
            
            if(map.containsKey(x)) map.put(x,map.get(x)+1);
            else map.put(x,1);
            
        }

        // Set<Character> vowelSet = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));
        String vowels = "aeiou";

        for(char x : map.keySet()) {
            
            if(vowels.indexOf(x)!=-1) maxV = Math.max(maxV,map.get(x));
            else maxC = Math.max(maxC,map.get(x));
        }

        return maxV+maxC;     
    }
}